subsys/caf/modules/sensor_sampler.c

```diff
@@ -180,25 +209,42 @@ static void trigger_handler(const struct device *dev, struct sensor_trigger *tri
 		return;
 	}
 
+	reset_sensor_trigger_counter(sc, sd);
 	update_sensor_state(sc, sd, SENSOR_STATE_ACTIVE);
+
+	if (IS_ENABLED(CONFIG_CAF_SENSOR_SAMPLER_ACTIVE_PM) && !wake_up_disabled) {
+		LOG_INF("Event (%d) from sensor %s triggers wake up",
+			trigger->type, sc->dev_name);
+		EVENT_SUBMIT(new_wake_up_event());
+	}
+
 	k_sem_give(&can_sample);
 }
 
-static void try_enter_sleep(const struct sensor_config *sc,
-			   struct sensor_data *sd,
-			   const float *curr)
+static void enter_sleep_if_configured(const struct sensor_config *sc,
```

Comment:
This function is called only if `trigger` is configured. No point to name it `*_if_configured`.